packages/teams-js/src/public/meeting.ts

```diff
@@ -925,6 +925,124 @@ export namespace meeting {
     );
   }
 
+  /**
+   * This function is used to join a meeting.
+   * This opens a meeting in a new window for the desktop app.
+   * In case of a web app, it will close the current app and open the meeting in the same tab.
+   * @param joinMeetingParams The parameters for joining the meeting. {@link JoinMeetingParams}
+   * @throws error if the meeting join fails, the promise will reject to an object with the error message.
+   */
+
+  export function joinMeeting(joinMeetingParams: JoinMeetingParams): Promise<void> {
+    if (!validateJoinMeetingParams(joinMeetingParams)) {
+      return Promise.reject('Invalid joinMeetingParams');
+    }
+    const { threadId, messageId, replyChainMessageId } = joinMeetingParams.chatInfo;
+
+    joinMeetingParams.joinWebUrl = new URL(joinMeetingParams.joinWebUrl);
+    const serializedJoinMeetingParams = {
+      joinWebUrl: joinMeetingParams.joinWebUrl.href,
+      subject: joinMeetingParams.subject,
+      source: joinMeetingParams.source || EventActionSource.Other,
+      chatInfo: {
+        threadId,
+        messageId: messageId || '0',
+        replyChainMessageId: replyChainMessageId || '0',
+      },
+    };
+    return new Promise<void>((resolve) => {
+      ensureInitialized(runtime);
+      resolve(
+        sendAndHandleStatusAndReason(
+          getApiVersionTag(ApiVersionNumber.V_2, ApiName.Meeting_JoinMeeting),
+          'meeting.joinMeeting',
+          serializedJoinMeetingParams,
+        ),
+      );
+    });
+  }
+
+  /**
+   * This function is used to check the validity of joinMeetingParams which return true if the joinMeetingParams' joinWebUrl and chatInfo are valid else false.
+   * @param joinMeetingParams The parameters for joining the meeting. {@link JoinMeetingParams}
+   * @returns false if joinMeetingParams' joinWebUrl or chatInfo is not valid else true.
+   */
+  function validateJoinMeetingParams(joinMeetingParams: JoinMeetingParams): boolean {
+    if (
+      !joinMeetingParams?.chatInfo?.threadId ||
+      !joinMeetingParams?.joinWebUrl ||
+      joinMeetingParams.chatInfo.threadId === ''
+    ) {
+      return false;
+    }
+    if (typeof joinMeetingParams.joinWebUrl === 'string' && joinMeetingParams.joinWebUrl === '') {
```

Comment:
Would it make sense to also check that the string is a valid URL here?

Reply:
@TrevorJoelHarris, we are converting the string to a url type in line 942:

> joinMeetingParams.joinWebUrl = new URL(joinMeetingParams.joinWebUrl); 

This essentially removes the possibility of string being an invalid url.

Reply:
A function called `validateJoinMeetingParams` should validate all of the parameters correctness, including that the joinUrl is a valid URL. If someone adds another function to this file that uses this helper function, they should be able to rely on this function validating *all* parameter correctness.

Reply:
@aditya-darji fixed this